supervise.go

```diff
@@ -197,6 +198,24 @@ func (s *service) setProcess(p *os.Process) {
 	s.process = p
 }
 
+func (s *service) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&struct {
+		Stopped   bool
+		StartTime time.Time
+		Attempt   uint64
+		Pid       int
+		Path      string
+		Args      []string
+	}{
+		Stopped:   s.stopped,
```

Comment:
You’ll need to acquire stoppedMu, startedMu and processMu to safely access these fields.

Also, I think the attempt field is not currently used actually — I can only see usage of a local `attempt` variable.